src/Compilers/VisualBasic/Portable/Lowering/IteratorRewriter/IteratorRewriter.vb

```diff
@@ -95,22 +96,25 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             EnsureSpecialType(SpecialType.System_Object, hasErrors)
             EnsureSpecialType(SpecialType.System_Boolean, hasErrors)
             EnsureSpecialType(SpecialType.System_Int32, hasErrors)
+            EnsureSpecialType(SpecialType.System_IDisposable, hasErrors)
 
-            If Me.Method.ReturnType.IsDefinition Then
-                If Me._isEnumerable Then
-                    EnsureSpecialType(SpecialType.System_Collections_IEnumerator, hasErrors)
-                End If
-            Else
-                If Me._isEnumerable Then
-                    EnsureSpecialType(SpecialType.System_Collections_Generic_IEnumerator_T, hasErrors)
-                    EnsureSpecialType(SpecialType.System_Collections_IEnumerable, hasErrors)
-                End If
-
-                EnsureSpecialType(SpecialType.System_Collections_IEnumerator, hasErrors)
+            ' IEnumerator
+            EnsureSpecialType(SpecialType.System_Collections_IEnumerator, hasErrors)
```

Comment:
Here, however, the check for the type isn't redundant because EnsureSpecialMember doesn't check use-site error on the containing type for the purpose of setting hasErrors to true. The diagnostics will be reported though.